<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: String objects representing biological sequences, and matching
algorithms</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="R.css">
</head><body>
<h1> String objects representing biological sequences, and matching
algorithms
<img class="toplogo" src="../../../doc/html/logo.jpg" alt="[R logo]">
</h1>
<hr>
<div align="center">
<a href="../../../doc/html/packages.html"><img src="../../../doc/html/left.jpg" alt="[Up]" width="30" height="30" border="0"></a>
<a href="../../../doc/html/index.html"><img src="../../../doc/html/up.jpg" alt="[Top]" width="30" height="30" border="0"></a>
</div><h2>Documentation for package &lsquo;Biostrings&rsquo; version 2.32.0</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">User guides, package vignettes and other documentation.</a></li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<p align="center">
<a href="#A">A</a>
<a href="#B">B</a>
<a href="#C">C</a>
<a href="#D">D</a>
<a href="#E">E</a>
<a href="#F">F</a>
<a href="#G">G</a>
<a href="#H">H</a>
<a href="#I">I</a>
<a href="#L">L</a>
<a href="#M">M</a>
<a href="#N">N</a>
<a href="#O">O</a>
<a href="#P">P</a>
<a href="#Q">Q</a>
<a href="#R">R</a>
<a href="#S">S</a>
<a href="#T">T</a>
<a href="#U">U</a>
<a href="#V">V</a>
<a href="#W">W</a>
<a href="#X">X</a>
<a href="#Y">Y</a>
<a href="#misc">misc</a>
</p>


<h2><a name="A">-- A --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="MultipleAlignment-class.html">AAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">AAMultipleAlignment-class</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="AAString-class.html">AAString</a></td>
<td>AAString objects</td></tr>
<tr><td width="25%"><a href="AAString-class.html">AAString-class</a></td>
<td>AAString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">AAStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">AAStringSet-class</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">AAStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">AAStringSetList-class</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="AAString-class.html">AA_ALPHABET</a></td>
<td>AAString objects</td></tr>
<tr><td width="25%"><a href="AAString-class.html">AA_PROTEINOGENIC</a></td>
<td>AAString objects</td></tr>
<tr><td width="25%"><a href="AAString-class.html">AA_STANDARD</a></td>
<td>AAString objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">ACtree2</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">ACtree2-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="align-utils.html">align-utils</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">aligned</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">aligned-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">aligned-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">AlignedXStringSet</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">AlignedXStringSet-class</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">AlignedXStringSet0</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">AlignedXStringSet0-class</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">alphabet</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">alphabet-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">alphabet-method</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">alphabetFrequency</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">alphabetFrequency-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">alphabetFrequency-method</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="AMINO_ACID_CODE.html">AMINO_ACID_CODE</a></td>
<td>The Single-Letter Amino Acid Code</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">append-method</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">as.character-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">as.character-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">as.character-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">as.character-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">as.character-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">as.character-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">as.character-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">as.integer-method</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">as.list-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">as.matrix-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">as.matrix-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">as.matrix-method</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">as.matrix-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">as.matrix-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">as.numeric-method</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">as.vector-method</a></td>
<td>XStringSet objects</td></tr>
</table>

<h2><a name="B">-- B --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="substitution_matrices.html">BLOSUM100</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">BLOSUM45</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">BLOSUM50</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">BLOSUM62</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">BLOSUM80</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">BOC2_SubjectString</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">BOC2_SubjectString-class</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">BOC_SubjectString</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">BOC_SubjectString-class</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">BString</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">BString-class</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">BStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">BStringSet-class</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">BStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">BStringSetList-class</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">ByPos_MIndex-class</a></td>
<td>MIndex objects</td></tr>
</table>

<h2><a name="C">-- C --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="translate.html">cDNA</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="chartr.html">chartr-method</a></td>
<td>Translating letters of a sequence</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">class:AAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="AAString-class.html">class:AAString</a></td>
<td>AAString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">class:AAStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">class:AAStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:ACtree2</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">class:AlignedXStringSet</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">class:AlignedXStringSet0</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">class:BOC2_SubjectString</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">class:BOC_SubjectString</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">class:BString</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">class:BStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">class:BStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">class:ByPos_MIndex</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">class:DNAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="DNAString-class.html">class:DNAString</a></td>
<td>DNAString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">class:DNAStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">class:DNAStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:Expanded_TB_PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">class:IlluminaQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="InDel-class.html">class:InDel</a></td>
<td>InDel objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">class:MaskedAAString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">class:MaskedBString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">class:MaskedDNAString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">class:MaskedRNAString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">class:MaskedXString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">class:MIndex</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:MTB_PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">class:MultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">class:PairwiseAlignments</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">class:PairwiseAlignmentsSingleSubject</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">class:PairwiseAlignmentsSingleSubjectSummary</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:PDict3Parts</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">class:PhredQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:PreprocessedTB</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">class:QualityAlignedXStringSet</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">class:QualityScaledAAStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">class:QualityScaledBStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">class:QualityScaledDNAStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">class:QualityScaledRNAStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">class:QualityScaledXStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">class:RNAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="RNAString-class.html">class:RNAString</a></td>
<td>RNAString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">class:RNAStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">class:RNAStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">class:SolexaQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:TB_PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">class:Twobit</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">class:XString</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">class:XStringPartialMatches</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">class:XStringQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">class:XStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">class:XStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">class:XStringViews</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="translate.html">codons</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="translate.html">codons-method</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">coerce-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">coerce-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">coerce-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">coerce-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">coerce-method</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">coerce-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">coerce-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">collapse-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">colmask</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">colmask-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">colmask&lt;-</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">colmask&lt;--method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="XStringSet-comparison.html">compare-method</a></td>
<td>Comparing and ordering the elements in one or more XStringSet objects</td></tr>
<tr><td width="25%"><a href="align-utils.html">compareStrings</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="align-utils.html">compareStrings-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="reverseComplement.html">complement</a></td>
<td>Sequence reversing and complementing</td></tr>
<tr><td width="25%"><a href="reverseComplement.html">complement-method</a></td>
<td>Sequence reversing and complementing</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">complementedPalindromeArmLength</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">complementedPalindromeArmLength-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">complementedPalindromeLeftArm</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">complementedPalindromeLeftArm-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">complementedPalindromeRightArm</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">complementedPalindromeRightArm-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="PDict-class.html">computeAllFlinks</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">computeAllFlinks-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">consensusMatrix</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">consensusMatrix-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="align-utils.html">consensusMatrix-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">consensusMatrix-method</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">consensusString</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">consensusString-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">consensusString-method</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">consensusViews</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">consensusViews-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">countIndex</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">countIndex-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="matchPattern.html">countPattern</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">countPattern-method</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="matchPattern.html">countPattern-method</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">countPDict</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">countPDict-method</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPWM.html">countPWM</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">countPWM-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="align-utils.html">coverage-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="match-utils.html">coverage-method</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
</table>

<h2><a name="D">-- D --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="InDel-class.html">deletion</a></td>
<td>InDel objects</td></tr>
<tr><td width="25%"><a href="InDel-class.html">deletion-method</a></td>
<td>InDel objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">deletion-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="detail.html">detail</a></td>
<td>Show (display) detailed object content</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">detail-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">dim-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="nucleotideFrequency.html">dinucleotideFrequency</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
<tr><td width="25%"><a href="dinucleotideFrequencyTest.html">dinucleotideFrequencyTest</a></td>
<td>Pearson's chi-squared Test and G-tests for String Position Dependence</td></tr>
<tr><td width="25%"><a href="dinucleotideFrequencyTest.html">dinucleotideFrequencyTest-method</a></td>
<td>Pearson's chi-squared Test and G-tests for String Position Dependence</td></tr>
<tr><td width="25%"><a href="translate.html">dna2rna</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">DNAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">DNAMultipleAlignment-class</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="DNAString-class.html">DNAString</a></td>
<td>DNAString objects</td></tr>
<tr><td width="25%"><a href="DNAString-class.html">DNAString-class</a></td>
<td>DNAString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">DNAStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">DNAStringSet-class</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">DNAStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">DNAStringSetList-class</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="DNAString-class.html">DNA_ALPHABET</a></td>
<td>DNAString objects</td></tr>
<tr><td width="25%"><a href="DNAString-class.html">DNA_BASES</a></td>
<td>DNAString objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">duplicated-method</a></td>
<td>PDict objects</td></tr>
</table>

<h2><a name="E">-- E --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="MIndex-class.html">elementLengths-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">encoding</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">encoding-method</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">end-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">endIndex</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">endIndex-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">errorSubstitutionMatrices</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="PDict-class.html">Expanded_TB_PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">Expanded_TB_PDict-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">extractAllMatches</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="replaceAt.html">extractAt</a></td>
<td>Extract/replace arbitrary substrings from/in a string or set of strings.</td></tr>
<tr><td width="25%"><a href="replaceAt.html">extractAt-method</a></td>
<td>Extract/replace arbitrary substrings from/in a string or set of strings.</td></tr>
</table>

<h2><a name="F">-- F --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="XStringSet-io.html">fasta.info</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="XStringSet-io.html">fastq.geometry</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">findComplementedPalindromes</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">findComplementedPalindromes-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">findPalindromes</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">findPalindromes-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
</table>

<h2><a name="G">-- G --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="MaskedXString-class.html">gaps-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="GENETIC_CODE.html">GENETIC_CODE</a></td>
<td>The Standard Genetic Code and its known variants</td></tr>
<tr><td width="25%"><a href="GENETIC_CODE.html">GENETIC_CODE_TABLE</a></td>
<td>The Standard Genetic Code and its known variants</td></tr>
<tr><td width="25%"><a href="GENETIC_CODE.html">getGeneticCode</a></td>
<td>The Standard Genetic Code and its known variants</td></tr>
<tr><td width="25%"><a href="getSeq.html">getSeq</a></td>
<td>getSeq</td></tr>
<tr><td width="25%"><a href="gregexpr2.html">gregexpr2</a></td>
<td>A replacement for R standard gregexpr function</td></tr>
</table>

<h2><a name="H">-- H --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="PDict-class.html">hasAllFlinks</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">hasAllFlinks-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">hasLetterAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">hasOnlyBaseLetters</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">hasOnlyBaseLetters-method</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="PDict-class.html">head-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="HNF4alpha.html">HNF4alpha</a></td>
<td>Known HNF4alpha binding sequences</td></tr>
</table>

<h2><a name="I">-- I --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="XStringQuality-class.html">IlluminaQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">IlluminaQuality-class</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="InDel-class.html">InDel</a></td>
<td>InDel objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">indel</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="InDel-class.html">InDel-class</a></td>
<td>InDel objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">indel-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">indel-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">initialize-method</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">initialize-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="injectHardMask.html">injectHardMask</a></td>
<td>Injecting a hard mask in a sequence</td></tr>
<tr><td width="25%"><a href="injectHardMask.html">injectHardMask-method</a></td>
<td>Injecting a hard mask in a sequence</td></tr>
<tr><td width="25%"><a href="InDel-class.html">insertion</a></td>
<td>InDel objects</td></tr>
<tr><td width="25%"><a href="InDel-class.html">insertion-method</a></td>
<td>InDel objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">insertion-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">intersect-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">isMatchingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">isMatchingEndingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">isMatchingEndingAt-method</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">isMatchingStartingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">isMatchingStartingAt-method</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="IUPAC_CODE_MAP.html">IUPAC_CODE_MAP</a></td>
<td>The IUPAC Extended Genetic Alphabet</td></tr>
</table>

<h2><a name="L">-- L --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="pmatchPattern.html">lcprefix</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="pmatchPattern.html">lcprefix-method</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="pmatchPattern.html">lcsubstr</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="pmatchPattern.html">lcsubstr-method</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="pmatchPattern.html">lcsuffix</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="pmatchPattern.html">lcsuffix-method</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">length-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">length-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">length-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">length-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">length-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="letter.html">letter</a></td>
<td>Subsetting a string</td></tr>
<tr><td width="25%"><a href="letter.html">letter-method</a></td>
<td>Subsetting a string</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">letterFrequency</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">letterFrequency-method</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">letterFrequencyInSlidingView</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">letterFrequencyInSlidingView-method</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="longestConsecutive.html">longestConsecutive</a></td>
<td>Obtain the length of the longest substring containing only 'letter'</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">lowlevel-matching</a></td>
<td>Low-level matching functions</td></tr>
</table>

<h2><a name="M">-- M --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="maskMotif.html">mask</a></td>
<td>Masking by content (or by position)</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedAAString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedAAString-class</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedBString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedBString-class</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskeddim</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskeddim-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedDNAString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedDNAString-class</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskedncol</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskedncol-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskednrow</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskednrow-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">maskedratio-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskedratio-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedRNAString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedRNAString-class</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">maskedwidth-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedXString</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">MaskedXString-class</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskGaps</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskGaps-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="maskMotif.html">maskMotif</a></td>
<td>Masking by content (or by position)</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">maskMotif-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="maskMotif.html">maskMotif-method</a></td>
<td>Masking by content (or by position)</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">masks</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">masks-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">masks&lt;-</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">masks&lt;--method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-comparison.html">match-method</a></td>
<td>Comparing and ordering the elements in one or more XStringSet objects</td></tr>
<tr><td width="25%"><a href="match-utils.html">match-utils</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
<tr><td width="25%"><a href="matchLRPatterns.html">matchLRPatterns</a></td>
<td>Find paired matches in a sequence</td></tr>
<tr><td width="25%"><a href="matchLRPatterns.html">matchLRPatterns-method</a></td>
<td>Find paired matches in a sequence</td></tr>
<tr><td width="25%"><a href="matchPattern.html">matchPattern</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="BOC_SubjectString-class.html">matchPattern-method</a></td>
<td>BOC\_SubjectString and BOC2\_SubjectString objects</td></tr>
<tr><td width="25%"><a href="matchPattern.html">matchPattern-method</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">matchPDict</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">matchPDict-exact</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-inexact.html">matchPDict-inexact</a></td>
<td>Inexact matching with matchPDict()/countPDict()/whichPDict()</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">matchPDict-method</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchProbePair.html">matchProbePair</a></td>
<td>Find "theoretical amplicons" mapped to a probe pair</td></tr>
<tr><td width="25%"><a href="matchProbePair.html">matchProbePair-method</a></td>
<td>Find "theoretical amplicons" mapped to a probe pair</td></tr>
<tr><td width="25%"><a href="matchprobes.html">matchprobes</a></td>
<td>A function to match a query sequence to the sequences of a set of probes.</td></tr>
<tr><td width="25%"><a href="matchPWM.html">matchPWM</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">matchPWM-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">maxScore</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">maxScore-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">maxWeights</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">maxWeights-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="IUPAC_CODE_MAP.html">mergeIUPACLetters</a></td>
<td>The IUPAC Extended Genetic Alphabet</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">MIndex</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">MIndex-class</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="matchPWM.html">minScore</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">minScore-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">minWeights</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">minWeights-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="misc.html">misc</a></td>
<td>Some miscellaneous stuff</td></tr>
<tr><td width="25%"><a href="match-utils.html">mismatch</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
<tr><td width="25%"><a href="align-utils.html">mismatch-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="match-utils.html">mismatch-method</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
<tr><td width="25%"><a href="align-utils.html">mismatchSummary</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="align-utils.html">mismatchSummary-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="align-utils.html">mismatchTable</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="align-utils.html">mismatchTable-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="nucleotideFrequency.html">mkAllStrings</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
<tr><td width="25%"><a href="PDict-class.html">MTB_PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">MTB_PDict-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">MultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">MultipleAlignment-class</a></td>
<td>MultipleAlignment objects</td></tr>
</table>

<h2><a name="N">-- N --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="misc.html">N50</a></td>
<td>Some miscellaneous stuff</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">names-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">names-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">names&lt;--method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">names&lt;--method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">narrow-method</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">narrow-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">nchar-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">nchar-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">nchar-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">nchar-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">nchar-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">nchar-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">nchar-method</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">nchar-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">ncol-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="align-utils.html">nedit</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="align-utils.html">nedit-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">neditAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">neditEndingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">neditEndingAt-method</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">neditStartingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">neditStartingAt-method</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="needwunsQS.html">needwunsQS</a></td>
<td>(Deprecated) Needleman-Wunsch Global Alignment</td></tr>
<tr><td width="25%"><a href="needwunsQS.html">needwunsQS-method</a></td>
<td>(Deprecated) Needleman-Wunsch Global Alignment</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">nindel</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">nindel-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">nindel-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="match-utils.html">nmatch</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
<tr><td width="25%"><a href="align-utils.html">nmatch-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="match-utils.html">nmatch-method</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
<tr><td width="25%"><a href="match-utils.html">nmismatch</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
<tr><td width="25%"><a href="align-utils.html">nmismatch-method</a></td>
<td>Utility functions related to sequence alignment</td></tr>
<tr><td width="25%"><a href="match-utils.html">nmismatch-method</a></td>
<td>Utility functions operating on the matches returned by a high-level matching function</td></tr>
<tr><td width="25%"><a href="PDict-class.html">nnodes</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">nnodes-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">nrow-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="nucleotideFrequency.html">nucleotideFrequencyAt</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
<tr><td width="25%"><a href="nucleotideFrequency.html">nucleotideFrequencyAt-method</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">nucleotideSubstitutionMatrix</a></td>
<td>Scoring matrices</td></tr>
</table>

<h2><a name="O">-- O --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="nucleotideFrequency.html">oligonucleotideFrequency</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
<tr><td width="25%"><a href="nucleotideFrequency.html">oligonucleotideFrequency-method</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
<tr><td width="25%"><a href="nucleotideFrequency.html">oligonucleotideTransitions</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
</table>

<h2><a name="P">-- P --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="padAndClip.html">padAndClip</a></td>
<td>Pad and clip strings</td></tr>
<tr><td width="25%"><a href="pairwiseAlignment.html">pairwiseAlignment</a></td>
<td>Optimal Pairwise Alignment</td></tr>
<tr><td width="25%"><a href="pairwiseAlignment.html">pairwiseAlignment-method</a></td>
<td>Optimal Pairwise Alignment</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignments</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignments-class</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-io.html">PairwiseAlignments-io</a></td>
<td>Write a PairwiseAlignments object to a file</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignments-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignmentsSingleSubject</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignmentsSingleSubject-class</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignmentsSingleSubject-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignmentsSingleSubjectSummary</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">PairwiseAlignmentsSingleSubjectSummary-class</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">palindromeArmLength</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">palindromeArmLength-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">palindromeLeftArm</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">palindromeLeftArm-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">palindromeRightArm</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="findPalindromes.html">palindromeRightArm-method</a></td>
<td>Searching a sequence for palindromes or complemented palindromes</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">PAM120</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">PAM250</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">PAM30</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">PAM40</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">PAM70</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">partitioning</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">pattern</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">pattern-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">pattern-method</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">patternFrequency</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">patternFrequency-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">PDict-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">PDict-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">PDict3Parts</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">PDict3Parts-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="phiX174Phage.html">phiX174Phage</a></td>
<td>Versions of bacteriophage phiX174 complete genome and sample short reads</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">PhredQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">PhredQuality-class</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="pid.html">pid</a></td>
<td>Percent Sequence Identity</td></tr>
<tr><td width="25%"><a href="pid.html">pid-method</a></td>
<td>Percent Sequence Identity</td></tr>
<tr><td width="25%"><a href="pmatchPattern.html">pmatchPattern</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="pmatchPattern.html">pmatchPattern-method</a></td>
<td>Longest Common Prefix/Suffix/Substring searching functions</td></tr>
<tr><td width="25%"><a href="PDict-class.html">PreprocessedTB</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">PreprocessedTB-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="needwunsQS.html">print.needwunsQS</a></td>
<td>(Deprecated) Needleman-Wunsch Global Alignment</td></tr>
<tr><td width="25%"><a href="matchPWM.html">PWM</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">PWM-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="matchPWM.html">PWMscoreStartingAt</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
</table>

<h2><a name="Q">-- Q --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">quality</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">quality-method</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">QualityAlignedXStringSet</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">QualityAlignedXStringSet-class</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledAAStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledAAStringSet-class</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledBStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledBStringSet-class</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledDNAStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledDNAStringSet-class</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledRNAStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledRNAStringSet-class</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledXStringSet</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">QualityScaledXStringSet-class</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">qualitySubstitutionMatrices</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="phiX174Phage.html">quPhiX174</a></td>
<td>Versions of bacteriophage phiX174 complete genome and sample short reads</td></tr>
</table>

<h2><a name="R">-- R --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="MultipleAlignment-class.html">readAAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="XStringSet-io.html">readAAStringSet</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="XStringSet-io.html">readBStringSet</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">readDNAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="XStringSet-io.html">readDNAStringSet</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">readRNAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="XStringSet-io.html">readRNAStringSet</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">rep-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">rep-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="replaceAt.html">replaceAt</a></td>
<td>Extract/replace arbitrary substrings from/in a string or set of strings.</td></tr>
<tr><td width="25%"><a href="replaceAt.html">replaceAt-method</a></td>
<td>Extract/replace arbitrary substrings from/in a string or set of strings.</td></tr>
<tr><td width="25%"><a href="replaceLetterAt.html">replaceLetterAt</a></td>
<td>Replacing letters in a sequence (or set of sequences) at some specified locations</td></tr>
<tr><td width="25%"><a href="replaceLetterAt.html">replaceLetterAt-method</a></td>
<td>Replacing letters in a sequence (or set of sequences) at some specified locations</td></tr>
<tr><td width="25%"><a href="reverseComplement.html">reverse-method</a></td>
<td>Sequence reversing and complementing</td></tr>
<tr><td width="25%"><a href="reverseComplement.html">reverseComplement</a></td>
<td>Sequence reversing and complementing</td></tr>
<tr><td width="25%"><a href="matchPWM.html">reverseComplement-method</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="reverseComplement.html">reverseComplement-method</a></td>
<td>Sequence reversing and complementing</td></tr>
<tr><td width="25%"><a href="translate.html">rna2dna</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">RNAMultipleAlignment</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">RNAMultipleAlignment-class</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="RNAString-class.html">RNAString</a></td>
<td>RNAString objects</td></tr>
<tr><td width="25%"><a href="RNAString-class.html">RNAString-class</a></td>
<td>RNAString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">RNAStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">RNAStringSet-class</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">RNAStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">RNAStringSetList-class</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="RNAString-class.html">RNA_ALPHABET</a></td>
<td>RNAString objects</td></tr>
<tr><td width="25%"><a href="RNAString-class.html">RNA_BASES</a></td>
<td>RNAString objects</td></tr>
<tr><td width="25%"><a href="GENETIC_CODE.html">RNA_GENETIC_CODE</a></td>
<td>The Standard Genetic Code and its known variants</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">rowmask</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">rowmask-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">rowmask&lt;-</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">rowmask&lt;--method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">rownames-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">rownames&lt;--method</a></td>
<td>MultipleAlignment objects</td></tr>
</table>

<h2><a name="S">-- S --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="XStringSet-io.html">saveXStringSet</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">score-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">seqtype-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">seqtype-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">seqtype-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">seqtype-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">seqtype-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">seqtype-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">seqtype-method</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">seqtype-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">seqtype&lt;--method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">seqtype&lt;--method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">seqtype&lt;--method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">seqtype&lt;--method</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">seqtype&lt;--method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">setdiff-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">setequal-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">show-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">show-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">show-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">show-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">show-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">show-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">show-method</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">show-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">show-method</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">show-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">show-method</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">show-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">SolexaQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">SolexaQuality-class</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="phiX174Phage.html">srPhiX174</a></td>
<td>Versions of bacteriophage phiX174 complete genome and sample short reads</td></tr>
<tr><td width="25%"><a href="padAndClip.html">stackStrings</a></td>
<td>Pad and clip strings</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">start-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">startIndex</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">startIndex-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="stringDist.html">stringDist</a></td>
<td>String Distance/Alignment Score Matrix</td></tr>
<tr><td width="25%"><a href="stringDist.html">stringDist-method</a></td>
<td>String Distance/Alignment Score Matrix</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">subject-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">subpatterns</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">subpatterns-method</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">subseq-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">subseq-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">subseq&lt;--method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="substitution_matrices.html">substitution.matrices</a></td>
<td>Scoring matrices</td></tr>
<tr><td width="25%"><a href="XString-class.html">substr-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">substring-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">summary-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
</table>

<h2><a name="T">-- T --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="PDict-class.html">tail-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">tb</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">tb-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">tb.width</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">tb.width-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">TB_PDict</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">TB_PDict-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">threebands-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="toComplex.html">toComplex</a></td>
<td>Turning a DNA sequence into a vector of complex numbers</td></tr>
<tr><td width="25%"><a href="toComplex.html">toComplex-method</a></td>
<td>Turning a DNA sequence into a vector of complex numbers</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">toString-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">toString-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">toString-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">toString-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">toString-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">toString-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="translate.html">transcribe</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="translate.html">translate</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="translate.html">translate-method</a></td>
<td>Translating DNA/RNA sequences</td></tr>
<tr><td width="25%"><a href="trimLRPatterns.html">trimLRPatterns</a></td>
<td>Trim Flanking Patterns from Sequences</td></tr>
<tr><td width="25%"><a href="trimLRPatterns.html">trimLRPatterns-method</a></td>
<td>Trim Flanking Patterns from Sequences</td></tr>
<tr><td width="25%"><a href="nucleotideFrequency.html">trinucleotideFrequency</a></td>
<td>Calculate the frequency of oligonucleotides in a DNA or RNA sequence (and other related functions)</td></tr>
<tr><td width="25%"><a href="PDict-class.html">Twobit</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">Twobit-class</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">type</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">type-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
</table>

<h2><a name="U">-- U --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="AlignedXStringSet-class.html">unaligned</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">unaligned-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">union-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">uniqueLetters</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="letterFrequency.html">uniqueLetters-method</a></td>
<td>Calculate the frequency of letters in a biological sequence, or the consensus matrix of a set of sequences</td></tr>
<tr><td width="25%"><a href="matchPWM.html">unitScale</a></td>
<td>PWM creating, matching, and related utilities</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">unlist-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">unlist-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">unmasked</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">unmasked-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">unmasked-method</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">updateObject-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">updateObject-method</a></td>
<td>XStringSet objects</td></tr>
</table>

<h2><a name="V">-- V --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="matchPattern.html">vcountPattern</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="matchPattern.html">vcountPattern-method</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">vcountPDict</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">vcountPDict-method</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="MaskedXString-class.html">Views-method</a></td>
<td>MaskedXString objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">Views-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">Views-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="matchPattern.html">vmatchPattern</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="matchPattern.html">vmatchPattern-method</a></td>
<td>String searching functions</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">vmatchPDict</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">vmatchPDict-method</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">vwhichPDict</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">vwhichPDict-method</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
</table>

<h2><a name="W">-- W --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="lowlevel-matching.html">which.isMatchingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">which.isMatchingEndingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">which.isMatchingEndingAt-method</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">which.isMatchingStartingAt</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="lowlevel-matching.html">which.isMatchingStartingAt-method</a></td>
<td>Low-level matching functions</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">whichPDict</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="matchPDict-exact.html">whichPDict-method</a></td>
<td>Matching a dictionary of patterns against a reference</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">width-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">width-method</a></td>
<td>PDict objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">width-method</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">width0</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">width0-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="MultipleAlignment-class.html">write.phylip</a></td>
<td>MultipleAlignment objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-io.html">writePairwiseAlignments</a></td>
<td>Write a PairwiseAlignments object to a file</td></tr>
<tr><td width="25%"><a href="XStringSet-io.html">writeXStringSet</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="phiX174Phage.html">wtPhiX174</a></td>
<td>Versions of bacteriophage phiX174 complete genome and sample short reads</td></tr>
</table>

<h2><a name="X">-- X --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="xscat.html">xscat</a></td>
<td>Concatenate sequences contained in XString, XStringSet and/or XStringViews objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">XString</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XString-class.html">XString-class</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">XStringPartialMatches-class</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">XStringQuality</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringQuality-class.html">XStringQuality-class</a></td>
<td>PhredQuality, SolexaQuality and IlluminaQuality objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">XStringSet</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-class.html">XStringSet-class</a></td>
<td>XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-comparison.html">XStringSet-comparison</a></td>
<td>Comparing and ordering the elements in one or more XStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringSet-io.html">XStringSet-io</a></td>
<td>Read/write an XStringSet object from/to a file</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">XStringSetList</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringSetList-class.html">XStringSetList-class</a></td>
<td>XStringSetList objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">XStringViews</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">XStringViews-class</a></td>
<td>The XStringViews class</td></tr>
</table>

<h2><a name="Y">-- Y --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="yeastSEQCHR1.html">yeastSEQCHR1</a></td>
<td>An annotation data file for CHR1 in the yeastSEQ package</td></tr>
</table>

<h2><a name="misc">-- misc --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="XString-class.html">!=-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">!=-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="replaceLetterAt.html">.inplaceReplaceLetterAt</a></td>
<td>Replacing letters in a sequence (or set of sequences) at some specified locations</td></tr>
<tr><td width="25%"><a href="XString-class.html">==-method</a></td>
<td>BString objects</td></tr>
<tr><td width="25%"><a href="XStringViews-class.html">==-method</a></td>
<td>The XStringViews class</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">[-method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">[-method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="QualityScaledXStringSet-class.html">[-method</a></td>
<td>QualityScaledBStringSet, QualityScaledDNAStringSet, QualityScaledRNAStringSet and QualityScaledAAStringSet objects</td></tr>
<tr><td width="25%"><a href="XStringPartialMatches-class.html">[-method</a></td>
<td>XStringPartialMatches objects</td></tr>
<tr><td width="25%"><a href="AlignedXStringSet-class.html">[&lt;--method</a></td>
<td>AlignedXStringSet and QualityAlignedXStringSet objects</td></tr>
<tr><td width="25%"><a href="PairwiseAlignments-class.html">[&lt;--method</a></td>
<td>PairwiseAlignments, PairwiseAlignmentsSingleSubject, and PairwiseAlignmentsSingleSubjectSummary objects</td></tr>
<tr><td width="25%"><a href="MIndex-class.html">[[-method</a></td>
<td>MIndex objects</td></tr>
<tr><td width="25%"><a href="PDict-class.html">[[-method</a></td>
<td>PDict objects</td></tr>
</table>
</body></html>
