<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: Scoring matrices</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="R.css">
</head><body>

<table width="100%" summary="page for substitution.matrices {Biostrings}"><tr><td>substitution.matrices {Biostrings}</td><td align="right">R Documentation</td></tr></table>

<h2>Scoring matrices</h2>

<h3>Description</h3>

<p>Predefined substitution matrices for nucleotide and amino acid alignments.
</p>


<h3>Usage</h3>

<pre>
data(BLOSUM45)
data(BLOSUM50)
data(BLOSUM62)
data(BLOSUM80)
data(BLOSUM100)
data(PAM30)
data(PAM40)
data(PAM70)
data(PAM120)
data(PAM250)
nucleotideSubstitutionMatrix(match = 1, mismatch = 0, baseOnly = FALSE, type = "DNA")
qualitySubstitutionMatrices(fuzzyMatch = c(0, 1), alphabetLength = 4L, qualityClass = "PhredQuality", bitScale = 1)
errorSubstitutionMatrices(errorProbability, fuzzyMatch = c(0, 1), alphabetLength = 4L, bitScale = 1)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>match</code></td>
<td>
<p>the scoring for a nucleotide match.</p>
</td></tr>
<tr valign="top"><td><code>mismatch</code></td>
<td>
<p>the scoring for a nucleotide mismatch.</p>
</td></tr>
<tr valign="top"><td><code>baseOnly</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code>. If <code>TRUE</code>, only
uses the letters in the &quot;base&quot; alphabet i.e. &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;.</p>
</td></tr>
<tr valign="top"><td><code>type</code></td>
<td>
<p>either &quot;DNA&quot; or &quot;RNA&quot;.</p>
</td></tr>
<tr valign="top"><td><code>fuzzyMatch</code></td>
<td>
<p>a named or unnamed numeric vector representing the base
match probability.</p>
</td></tr>
<tr valign="top"><td><code>errorProbability</code></td>
<td>
<p>a named or unnamed numeric vector representing the error
probability.</p>
</td></tr>
<tr valign="top"><td><code>alphabetLength</code></td>
<td>
<p>an integer representing the number of letters
in the underlying string alphabet. For DNA and RNA, this would be 4L.
For Amino Acids, this could be 20L.</p>
</td></tr>
<tr valign="top"><td><code>qualityClass</code></td>
<td>
<p>a character string of <code>"PhredQuality"</code>,
<code>"SolexaQuality"</code>, or <code>"IlluminaQuality"</code>.</p>
</td></tr>
<tr valign="top"><td><code>bitScale</code></td>
<td>
<p>a numeric value to scale the quality-based substitution
matrices. By default, this is 1, representing bit-scale scoring.</p>
</td></tr>
</table>


<h3>Format</h3>

<p>The BLOSUM and PAM matrices are square symmetric matrices with integer
coefficients, whose row and column names are identical and unique: each
name is a single letter representing a nucleotide or an amino acid.
</p>
<p><code>nucleotideSubstitutionMatrix</code> produces a substitution matrix for all IUPAC
nucleic acid codes based upon match and mismatch parameters.
</p>
<p><code>errorSubstitutionMatrices</code> produces a two element list of numeric
square symmetric matrices, one for matches and one for mismatches.
</p>
<p><code>qualitySubstitutionMatrices</code> produces the substitution matrices
for Phred or Solexa quality-based reads.
</p>


<h3>Details</h3>

<p>The BLOSUM and PAM matrices are not unique. For example, the definition of
the widely used BLOSUM62 matrix varies depending on the source, and even a
given source can provide different versions of &quot;BLOSUM62&quot; without keeping
track of the changes over time. NCBI provides many matrices here
ftp://ftp.ncbi.nih.gov/blast/matrices/ but their definitions don't match
those of the matrices bundled with their stand-alone BLAST software
available here ftp://ftp.ncbi.nih.gov/blast/
</p>
<p>The BLOSUM45, BLOSUM62, BLOSUM80, PAM30 and PAM70 matrices were taken
from NCBI stand-alone BLAST software.
</p>
<p>The BLOSUM50, BLOSUM100, PAM40, PAM120 and PAM250 matrices were taken
from ftp://ftp.ncbi.nih.gov/blast/matrices/
</p>
<p>The quality matrices computed in <code>qualitySubstitutionMatrices</code> are
based on the paper by Ketil Malde. Let <i>&epsilon;_i</i> be the probability
of an error in the base read. For <code>"Phred"</code> quality measures <i>Q</i>
in <i>[0, 99]</i>, these error probabilities are given by
<i>&epsilon;_i = 10^{-Q/10}</i>. For <code>"Solexa"</code> quality measures
<i>Q</i> in <i>[-5, 99]</i>, they are given by
<i>&epsilon;_i = 1 - 1/(1 + 10^{-Q/10})</i>.
Assuming independence within and between base reads, the combined error
probability of a mismatch when the underlying bases do match is
<i>&epsilon;_c = &epsilon;_1 + &epsilon;_2 - (n/(n-1)) * &epsilon;_1 * &epsilon;_2</i>,
where <i>n</i> is the number of letters in the underlying alphabet. Using
<i>&epsilon;_c</i>, the substitution score is given by when two bases match is given by
<i>b * \log_2(&gamma;_{x,y} * (1 - &epsilon;_c) * n + (1 - &gamma;_{x,y}) * &epsilon;_c * (n/(n-1)))</i>,
where <i>b</i> is the bit-scaling for the scoring and <i>&gamma;_{x,y}</i> is the
probability that characters <i>x</i> and <i>y</i> represents the same underlying
information (e.g. using IUPAC, <i>&gamma;_{A,A} = 1</i> and <i>&gamma;_{A,N} = 1/4</i>.
In the arguments listed above <code>fuzzyMatch</code> represents <i>&gamma;_{x,y}</i>
and <code>errorProbability</code> represents <i>&epsilon;_i</i>.
</p>


<h3>Author(s)</h3>

<p>H. Pages and P. Aboyoun</p>


<h3>References</h3>

<p>K. Malde, The effect of sequence quality on sequence alignment, Bioinformatics, Feb 23, 2008.
</p>


<h3>See Also</h3>

<p><code><a href="pairwiseAlignment.html">pairwiseAlignment</a></code>,
<a href="PairwiseAlignments-class.html">PairwiseAlignments-class</a>,
<a href="DNAString-class.html">DNAString-class</a>,
<a href="AAString-class.html">AAString-class</a>,
<a href="XStringQuality-class.html">PhredQuality-class</a>,
<a href="XStringQuality-class.html">SolexaQuality-class</a>,
<a href="XStringQuality-class.html">IlluminaQuality-class</a>
</p>


<h3>Examples</h3>

<pre>
  s1 &lt;- 
    DNAString("ACTTCACCAGCTCCCTGGCGGTAAGTTGATCAAAGGAAACGCAAAGTTTTCAAG")
  s2 &lt;-
    DNAString("GTTTCACTACTTCCTTTCGGGTAAGTAAATATATAAATATATAAAAATATAATTTTCATC")

  ## Fit a global pairwise alignment using edit distance scoring
  pairwiseAlignment(s1, s2,
                    substitutionMatrix = nucleotideSubstitutionMatrix(0, -1, TRUE),
                    gapOpening = 0, gapExtension = -1)

  ## Examine quality-based match and mismatch bit scores for DNA/RNA
  ## strings in pairwiseAlignment.
  ## By default patternQuality and subjectQuality are PhredQuality(22L).
  qualityMatrices &lt;- qualitySubstitutionMatrices()
  qualityMatrices["22", "22", "1"]
  qualityMatrices["22", "22", "0"]

  pairwiseAlignment(s1, s2)

  ## Get the substitution scores when the error probability is 0.1
  subscores &lt;- errorSubstitutionMatrices(errorProbability = 0.1)
  submat &lt;- matrix(subscores[,,"0"], 4, 4)
  diag(submat) &lt;- subscores[,,"1"]
  dimnames(submat) &lt;- list(DNA_ALPHABET[1:4], DNA_ALPHABET[1:4])
  submat
  pairwiseAlignment(s1, s2, substitutionMatrix = submat)

  ## Align two amino acid sequences with the BLOSUM62 matrix
  aa1 &lt;- AAString("HXBLVYMGCHFDCXVBEHIKQZ")
  aa2 &lt;- AAString("QRNYMYCFQCISGNEYKQN")
  pairwiseAlignment(aa1, aa2, substitutionMatrix = "BLOSUM62", gapOpening = -3, gapExtension = -1)

  ## See how the gap penalty influences the alignment
  pairwiseAlignment(aa1, aa2, substitutionMatrix = "BLOSUM62", gapOpening = -6, gapExtension = -2)

  ## See how the substitution matrix influences the alignment
  pairwiseAlignment(aa1, aa2, substitutionMatrix = "BLOSUM50", gapOpening = -3, gapExtension = -1)

  if (interactive()) {
    ## Compare our BLOSUM62 with BLOSUM62 from ftp://ftp.ncbi.nih.gov/blast/matrices/
    data(BLOSUM62)
    BLOSUM62["Q", "Z"]
    file &lt;- "ftp://ftp.ncbi.nih.gov/blast/matrices/BLOSUM62"
    b62 &lt;- as.matrix(read.table(file, check.names=FALSE))
    b62["Q", "Z"]
  }
</pre>

<hr><div align="center">[Package <em>Biostrings</em> version 2.32.0 <a href="00Index.html">Index</a>]</div>
</body></html>
